src/lib/nfs-workarounds.c

```diff
@@ -167,18 +174,23 @@ static void nfs_flush_chown_uid(const char *path)
 		i_error("nfs_flush_chown_uid: stat(%s) failed: %m", path);
 		return;
 	}
-#endif
-	if (chown(path, uid, (gid_t)-1) < 0) {
```

Comment:
This should be in another commit, not same commit.

Reply:
Hello and thank you for taking the time to review this pull request, I would like to apply the fixes you have suggested here but I will need some clarification regarding which piece should be separated into another commit.

When making the commit I considered the entire change to be a singular compartmentalized fix which changed the Linux implementation to use stat + chmod and switched the non-Linux implementations to retain the old chown mechanism. 

Am I correct in my assumption that the change for Linux and change for non-Linux should be separated into individual commits?